### PR TITLE
Fix remaining boost-gradle issues

### PR DESCRIPTION
**Pull request by mattbsox**
- refactor Docker to use Anjum's code <- Adding this in a new PR
- use default Maven Central runtime (right now we're defaulting from DHE)
- add package extension (discussed in call)
- tests should run gradle build
- add classifier tests
- add package + docker tests
- use transitive Spring Framework dependencies for feature detection (this broke when we made this change for boost-maven)

### CODE REVIEW

File: boost-gradle/src/main/groovy/io/openliberty/boost/liberty/tasks/BoostPackageTask.groovy [hunk 1]
```diff
@@ -86,7 +86,7 @@ public class BoostPackageTask extends AbstractBoostTask {
                         dependsOn 'bootRepackage'
 
                         //Skipping if Docker is configured
-                        if (!isDockerConfigured()) {
+                        if (!isDockerConfigured() || isPackageConfigured()) {
```
Comment (chyt) [inline boost-gradle/src/main/groovy/io/openliberty/boost/liberty/tasks/BoostPackageTask.groovy]: Do you also need this check on line 64?

File: boost-gradle/src/main/groovy/io/openliberty/boost/liberty/tasks/BoostPackageTask.groovy [hunk 2]
```diff
@@ -101,7 +101,13 @@ public class BoostPackageTask extends AbstractBoostTask {
 
             //The task will perform this before any other task actions
             doFirst {
-                // boostPackage.archive = "${project.buildDir}/" + boostPackage.archive
+                
+                if (isPackageConfigured()) {
```
Comment (chyt) [inline boost-gradle/src/main/groovy/io/openliberty/boost/liberty/tasks/BoostPackageTask.groovy]: Don't nest if statements.

File: boost-gradle/src/main/groovy/io/openliberty/boost/liberty/tasks/BoostPackageTask.groovy [hunk 3]
```diff
@@ -166,7 +172,9 @@ public class BoostPackageTask extends AbstractBoostTask {
 
             // Find and add appropriate springBoot features
             List<String> featuresNeededForSpringBootApp = SpringBootUtil.getLibertyFeaturesForSpringBoot(springBootVersion,
-                    getSpringBootStarters(), new BoostLogger())
+                    getSpringBootDependencies(), new BoostLogger())
+
+            featuresNeededForSpringBootApp.add("servlet-4.0")
```
Comment (chyt) [inline boost-gradle/src/main/groovy/io/openliberty/boost/liberty/tasks/BoostPackageTask.groovy]: Looks like you forgot to take this out.

File: boost-gradle/src/main/groovy/io/openliberty/boost/liberty/tasks/BoostPackageTask.groovy [hunk 4]
```diff
@@ -251,7 +259,12 @@ public class BoostPackageTask extends AbstractBoostTask {
     //returns true if bootJar is using the same archiveName as jar
     private boolean shouldReplaceProjectArchive() {
         if (project.plugins.hasPlugin('java')) {
-            return project.jar.archiveName == project.bootJar.archiveName
+            if (springBootVersion.startsWith('2.')) {
+                return project.jar.archiveName == project.bootJar.archiveName
+            } 
+            // else if (springBootVersion.startsWith('1.')) {
```
Comment (chyt) [inline boost-gradle/src/main/groovy/io/openliberty/boost/liberty/tasks/BoostPackageTask.groovy]: You need this.

Comment (mattbsox) [reply boost-gradle/src/main/groovy/io/openliberty/boost/liberty/tasks/BoostPackageTask.groovy]: Looks like the 1.5 case is always gonna be the jar.archiveName.

**Comment by scottkurz** [comment]
Merged the latest changes to https://github.com/scottkurz/boost-maven/pull/4   so the existing history (and review) could be preserved.     Would be cleaner to rebase/squash this into one or a couple commits before ultimately merging into master.

**Pull request merged**